routetable/route_table.go

```diff
@@ -286,7 +314,9 @@ func (r *RouteTable) Apply() error {
 				continue
 			}
 			ifaceName := attrs.Name
-			if r.ifacePrefixRegexp.MatchString(ifaceName) {
+
+			//set the iface to dirty if either the rtProtocolFilter is specified OR the iface is managed by this RouteTable
```

Comment:
Don't we only want to do this if the protocol filter is specified _and_ matches the filter? Or am I misunderstanding?


Also nit: I'd add a space after the comment and use sentence case.

e.g. `// Set the iface` rather than `//set the iface`